source/Engine.cpp

```diff
@@ -1690,9 +1730,40 @@ void Engine::CalculateStep()
 
 
 
+// Move a ship. Can be called concurrently.
+void Engine::MoveShip(const shared_ptr<Ship> &ship, vector<mutex> &bufferMutexes, vector<list<Visual>> &visualsBuffer,
```

Comment:
```suggestion
void Engine::MoveShip(const shared_ptr<Ship> &ship, vector<mutex> &bufferMutexes, vector<list<Visual>> &visualsBuffer,
```
these arguments are quite a mouthful, thoughts on making a structure(public class) to hold them?